scripts/deploy-workbook.sh

```diff
@@ -0,0 +1,213 @@
+        "groupType": "template",
+        "loadFromTemplateId": "${summary_id}",
+        "items": []
+      },
+      "conditionalVisibility": {
+        "parameterName": "selectedTab",
+        "comparison": "isEqualTo",
+        "value": "Summary2"
+      },
+      "name": "summary group"
+    }
+EOS
+)
+escaped_replacement_text=$(printf '%s\n' "$tab_of_Summary" | sed 's:[\/&]:\\&:g;$!s/$/\\/')
+sed -i "s/\${tab_of_Summary}/$escaped_replacement_text/g" workbook.tpl.json
+
+
+link_of_Advisor=$(cat <<EOS
+         ,{
+            "id": "d983c7c7-b5a0-4245-86fa-52ac1266fb13",
+            "cellValue": "selectedTab",
+            "linkTarget": "parameter",
+            "linkLabel": "Azure Advisor",
+            "subTarget": "Advisor",
+            "style": "link"
+          }
+EOS
+)
+escaped_replacement_text=$(printf '%s\n' "$link_of_Advisor" | sed 's:[\/&]:\\&:g;$!s/$/\\/')
+sed -i "s/\${link_of_Advisor}/$escaped_replacement_text/g" workbook.tpl.json
+
+advisor_id=$(cat ReliabilityWorkbookAdvisor_id | tr -d '\"')
+tab_of_Advisor=$(cat <<EOS
+    ,{
+      "type": 12,
+      "content": {
+        "version": "NotebookGroup/1.0",
+        "groupType": "template",
+        "loadFromTemplateId": "${advisor_id}",
+        "items": []
+      },
+      "conditionalVisibility": {
+        "parameterName": "selectedTab",
+        "comparison": "isEqualTo",
+        "value": "Advisor"
+      },
+      "name": "Advisor"
+    }
+EOS
+)
+escaped_replacement_text=$(printf '%s\n' "$tab_of_Advisor" | sed 's:[\/&]:\\&:g;$!s/$/\\/')
+sed -i "s/\${tab_of_Advisor}/$escaped_replacement_text/g" workbook.tpl.json
+
+link_of_Export=$(cat <<EOS
+          ,{
+            "id": "0f548bfa-f959-4a25-a9ac-7c986be6d33b",
+            "cellValue": "selectedTab",
+            "linkTarget": "parameter",
+            "linkLabel": "Export",
+            "subTarget": "Export",
+            "style": "link"
+          }
+EOS
+)
+escaped_replacement_text=$(printf '%s\n' "$link_of_Export" | sed 's:[\/&]:\\&:g;$!s/$/\\/')
+sed -i "s/\${link_of_Export}/$escaped_replacement_text/g" workbook.tpl.json
+
+export_id=$(cat ReliabilityWorkbookExport_id | tr -d '\"')
+tab_of_Export=$(cat <<EOS
+    ,{
+      "type": 12,
+      "content": {
+        "version": "NotebookGroup/1.0",
+        "groupType": "template",
+        "loadFromTemplateId": "${export_id}",
+        "items": []
+      },
+      "conditionalVisibility": {
+        "parameterName": "selectedTab",
+        "comparison": "isEqualTo",
+        "value": "Export"
+      },
+      "name": "ExportStep"
+    }
+EOS
+)
+escaped_replacement_text=$(printf '%s\n' "$tab_of_Export" | sed 's:[\/&]:\\&:g;$!s/$/\\/')
+sed -i "s/\${tab_of_Export}/$escaped_replacement_text/g" workbook.tpl.json
+
+log "Deploy FTA - Reliability Workbook"
+az deployment group create -g $resource_group_name --template-uri $base_url/artifacts/azuredeploy.json --parameters name="FTA - Reliability Workbook" serializedData=@workbook.tpl.json
```

Comment:
> We need to skip deployment of ReliabilityWorkbookPublic by default. Maybe we can keep this optional (with additional parameters passed to enable) for testing public version during development. But for end users, this should not be deployed, otherwise it will cause confusion.

Sorry, could you tell me which part of the ReliabilityWorkbookPublic you thought would be deployed?
$base_url/artifacts/azuredeploy.json is following:
https://github.com/tsubasaxZZZ/reliability-workbook-0313/blob/main/artifacts/azuredeploy.json
It just deploy top page of Community edition.
![image](https://user-images.githubusercontent.com/3822284/235594768-0278e5e6-511e-42a6-ab0b-46f13ead8d69.png)


Following line will deploy top page.
`az deployment group create -g $resource_group_name --template-uri $base_url/artifacts/azuredeploy.json --parameters name="FTA - Reliability Workbook" serializedData=@workbook.tpl.json`

Reply:
Thanks. Let me double check. Last time when I deployed, I seen both Public version as well in the same RG. Also, its good to cleanup all the additional files generated (_id, file list, etc.)

Reply:
@tsubasaxZZZ  I see only the community version now, thanks!

Reply:
> Also, can we cleanup the _id files at the script end once the workbooks are deployed? If we want to re-run for another deployment, the id might change, so we might not need to retain that data. Is my understanding right?

Yes, we can clean up the _id file.
I wanted to prevent having to start all over again when we run the tool again if for some reason the tool stops in the middle.
However, the ARM mechanism ensures equality even if the same deployment is done repeatedly. The current checks have no benefit other than to save time to be aware of the existence of the deployed workbook.
So I don't think it is a problem to remove it at the end.

So, let's remove this part and delete all _id files at the end of script:
```shell
  filename_base=`basename $f .workbook`
  filename=`basename $f`
  # file_size=$(stat -c%s "$f") <--------------------------------
  # if [ ! -e ${filename_base}_id -o $file_size -eq 0 ]; then <--------------------------------
    log "Deploy ${filename_base}"
    { az deployment group create --name $filename_base -g $resource_group_name --template-uri $base_url/artifacts/azuredeploy.json --parameters serializedData=@$filename --parameters name=$filename_base --query 'properties.outputs.resource_id.value' -o json; } > ${filename_base}_id &
  # fi <--------------------------------
```

Reply:
@tsubasaxZZZ I see bunch of files under scripts directory after execution. 
![image](https://github.com/Azure/reliability-workbook/assets/11145865/d17bded2-3abe-4700-9402-6f7422048839)

Also I see the deployment scripts pulls from artifacts folder and not scripts folder. 

`[ ! -f workbook_filelist ] && wget $base_url/artifacts/workbook_filelist
`
Hence we can cleanup all the files after moving required ones to artifacts. Also since we start maintaining file list to artifacts, workbook folder become redundant. So I would suggest we reuse workbook folder and have the generated workbooks there instead of workbooks. Also, whichever files we don't do a cleanup due to resuming failed job, we need to update .gitignore files for those paths.